lib/rails_multisite/connection_management.rb

```diff
@@ -245,10 +245,16 @@ def establish_connection(opts)
           @connection_handlers[handler_key(spec)] = handler
         end
       else
-        handler = @default_connection_handler
+        handler = ActiveRecord::Base.default_connection_handler
+
         if !@established_default
-          handler_establish_connection(handler, spec)
-          @established_default = true
+          begin
+            pool = handler_establish_connection(handler, spec)
+            pool.checkout
```

Comment:
the checkout here concerns me... why is it needed? how did it work in the past without it?

Reply:
So `establish_connection` is a little misleading. It creates a connection pool but doesn't actually make any connections to the database. Right here, I'm making a connection so that we can check if the connection is good and if not switch to the current connection handler. This is useful when booting a rails app against a Readonly PG. I have an initializer that ensures we boot with the reading connection handler.

https://github.com/discourse/rails_failover/blob/850d04f4b1c4178862d9674ec0b2a46a2f1c20d9/lib/rails_failover/active_record/railtie.rb#L22-L29

Reply:
but this means that our "establish_connection" works differently to how the rails one works, it looks like a bit of a trap.

Reply:
The only difference is that we'll blow up earlier if a connection cannot be made which isn't a bad idea.

Reply:
my bigger concern is that maybe there will be cases where connections will not be checked back in properly? 

Also ... can't this just be enabled in rails proper? Tell it that when you establish a connection you always want it to be established?

Reply:
Actually......the gem shouldn't have to established the connection here. Rails already handles that so we're actually doing double work here.

Reply:
O but I still need this so that we have a fallback. In Rails 6, `ActiveRecord::Base.default_connection_handler` is the default connection handler. This allows me to assign `ActiveRecord::Base.connection_handler` to a reading connection handler on boot without that being the default connection handler for rails_failover.